### PR TITLE
enabling changes or monitoring

### PR DESCRIPTION
**Pull request by punituee**
reporting latency on block_device_mapping, building, spawning and deleting the instance

### CODE REVIEW

File: nova/compute/api.py [hunk 1]
```diff
@@ -76,9 +76,13 @@
 from nova import utils
 from nova.virt import hardware
 from nova import volume
+from metricgenerator import publish
+from metricgenerator.logger import Logger
 
 LOG = logging.getLogger(__name__)
 
+publish = publish.Publish("nova-api", "/tmp/config.cfg")
```
Comment (abhishek-mehrotra) [inline nova/compute/api.py:83]: Pls put  /tmp/config.cfg file or take it from proper location

File: nova/compute/manager.py [hunk 2]
```diff
@@ -94,6 +97,7 @@
 from nova import volume
 from nova.volume import encryptors
 
+publish = publish.Publish("nova-compute", "/tmp/config1.cfg")
```
Comment (abhishek-mehrotra) [inline nova/compute/manager.py:99]: Pls put /tmp/config.cfg file or take it from proper location

**Comment by punituee** [comment]
cat /tmp/config.cfg 
[Constants]
service = nova-api
[Constants]
logdir = /tmp/metricgenerator/logs
[Constants]
filename = metric.log
[Constants]
socket = tcp://127.0.0.1:5444
[Constants]
hostname = kilo-v2-setup
[Constants]
bucket = asmitatest-metricgenerator

cat /tmp/config1.cfg 
[Constants]
service = nova-compute
[Constants]
logdir = /tmp/metricgenerator/logs
[Constants]
filename = metric.log
[Constants]
socket = tcp://127.0.0.1:5445
[Constants]
hostname = kilo-v2-setup
[Constants]
bucket = asmitatest-metricgenerator

For now we are putting these in tmp folder only. The two files are below.
I have created a jira bug which stops me from implementing a more generic solution.
https://jira.ril.com/browse/JCC-79

File: nova/compute/api.py [hunk 3]
```diff
@@ -76,9 +76,13 @@
 from nova import utils
 from nova.virt import hardware
 from nova import volume
+from metricgenerator import publish
+from metricgenerator.logger import Logger
```
Comment (varunarya10) [inline nova/compute/api.py]: Please remove this as it is not being used.

File: nova/compute/manager.py [hunk 4]
```diff
@@ -34,6 +34,9 @@
 import traceback
 import uuid
 
+from metricgenerator import publish
+from metricgenerator.logger import Logger
```
Comment (varunarya10) [inline nova/compute/manager.py]: Logger is not being used.

File: nova/compute/api.py [hunk 5]
```diff
@@ -1501,6 +1504,7 @@ def _check_multiple_instances_and_specified_ip(self, requested_networks):
                 raise exception.InvalidFixedIpAndMaxCountRequest(reason=msg)
 
     @hooks.add_hook("create_instance")
+    @publish.ReportLatency("block_device_mapping", listOfKeys = [[],['request_id']])
```
Comment (upadhyay-prashant) [inline nova/compute/api.py:1507]: this will not give you the creation time for BDM. if that is the motive here.

**Pull request merged**